pkg/util/ratelimiter/rate_limiter.go

```diff
@@ -0,0 +1,27 @@
+package ratelimiter
+
+import (
+	"flag"
+	"time"
+
+	"golang.org/x/time/rate"
+	"k8s.io/client-go/util/workqueue"
+)
+
+func init() {
+	flag.DurationVar(&baseDelay, "rate-limiter-base-delay", time.Millisecond*5, "The base delay for rate limiter. Defaults 5ms")
+	flag.DurationVar(&maxDelay, "rate-limiter-max-delay", time.Second*1000, "The max delay for rate limiter. Defaults 1000s")
+	flag.IntVar(&qps, "rate-limiter-qps", 10, "The qps for rate limier. Defaults 10")
+	flag.IntVar(&bucketSize, "rate-limiter-bucket-size", 100, "The bucket size for rate limier. Defaults 100")
+}
+
+var baseDelay, maxDelay time.Duration
+var qps, bucketSize int
+
+// DefaultControllerRateLimiter provide a default rate limiter for controller, and users can tune it by corresponding flags.
+func DefaultControllerRateLimiter() workqueue.RateLimiter {
```

Comment:
It is the reason why I'm not like putting flags to `init()`, when you want a default rate limiter, the flags will be applied to your system.